### ISSUE TITLE
sticky windows, in the sense of edges resistance+attraction

### ISSUE THREAD

**Issue by Arnaudv6** [WayfireWM/wayfire#1074]
as seen here for instance
https://www.youtube.com/watch?v=lhH5oNLw_4w

**Comment by ammen99** [comment]
I think this is out-of-scope for the main plugins. Here's how to implement it in a separate plugin (it would require some special-casing to get it working right with all kinds of plugins):

- Use https://github.com/WayfireWM/wayfire/blob/master/plugins/common/wayfire/plugins/common/move-drag-interface.hpp#L41 to figure out when a view is being dragged on a certain output
- There are currently not many plugins which support dragging. Expo, Scale and Move. You could check whether Expo and Scale are active and in these cases, not do anything.
- If just move is active, then:
  - Connect to `pointer_motion` signal.
  - Modify the motion signals so that the view is snapped to the edge
  - Eventually, after reaching a threshold or whatever, add all of the ignored x/y deltas for the next motion event (optional, idk how you want it to work).

Basically, your plugin is constraining the mouse to the correct regions, so that the move code does the correct thing. From what I saw in the video, this is also how compiz works.

An example of how to manipulate the pointer motion signal: https://github.com/WayfireWM/wayfire-plugins-extra/blob/master/src/force-fullscreen.cpp#L524

Feel free to ask further questions if you decide to implement this (but it won't be merged upstream).
